File: 026_remove_duplicates_from_sorted_array/Solution.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums == null || nums.length == 0) return 0;

        int j = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != nums[j])
                nums[++j] = nums[i];
        }

        return j + 1;
    }

    public static void main(String[] args) {
        int[] nums = new int[]{1, 1, 2, 2};
        Solution solution = new Solution();
        int newLen = solution.removeDuplicates(nums);
        for (int i = 0; i < newLen; i++) {
            System.out.print(nums[i] + " ");
        }
    }
}

File: 026_remove_duplicates_from_sorted_array/Solution.py
class Solution:
    def removeDuplicates(self, nums: list) -> int:
        if not nums or len(nums) == 0: return 0

        j = 0
        for i in range(len(nums)):
            if nums[i] != nums[j]:
                j += 1
                nums[j] = nums[i]

        return j + 1


if __name__ == '__main__':
    nums = [1, 1, 2, 2]
    solution = Solution()
    new_len = solution.removeDuplicates(nums)
    for i in range(new_len):
        print(nums[i], end = ' ')
